unit/memory-analyzer/gdb_api.cpp

```diff
@@ -0,0 +1,153 @@
+/*******************************************************************\
+
+Module: GDB Machine Interface API unit tests
```

Comment:
Is it really worth having a unit test here? It might be better to just run this in `regression/` as an integration test? There isn't a whole lot to the API, and the unit test is fairly close to being a shell script written in C++?

Reply:
The back and forth interaction with `gdb` is actually quite complicated. I feel like there definitely need to be unit tests to test that the API does this correctly.

Reply:
Ok, fine with me. I was just wondering whether doing this as a unit test just made things more complicated than necessary.